service/csvService.go

```diff
@@ -0,0 +1,138 @@
+package service
+
+import (
+	"encoding/csv"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+
+	"github.com/fatih/structs"
+)
+
+type CsvService interface {
+	FilterById(string) ([]Doc, error)
+	DownloadCsvData() error
+}
+
+type CsvServiceImpl struct{}
+
+type SearchResponse struct {
+	Doc      []Doc `json:"docs"`
+	NumFound int   `json:"numFound"`
+	Start    int   `json:"start"`
+}
+
+type Doc struct {
+	Key       string `json:"key"`
+	Title     string `json:"title"`
+	Type      string `json:"type"`
+	Published string `json:"first_published_year"`
+}
+
+var CSVFile = "/home/omar/Desktop/saved.csv"
+
+// func (c )LoadCsvData
+func (c CsvServiceImpl) DownloadCsvData() error {
+	getAndSaveData()
+	return nil
+}
+
+func (c CsvServiceImpl) FilterById(id string) ([]Doc, error) {
+	file, err := os.Open(CSVFile)
+	if err != nil {
+		return nil, err
```

Comment:
When working with external packages, it's better to wrap the error to add a specific error message in this layer.

```suggestion
		return nil, fmt.Errorf("failed to open file: %v", err)
```

Reply:
:+1: